proto/internal/uber/cadence/admin/v1/service.proto

```diff
@@ -0,0 +1,286 @@
+  rpc GetWorkflowExecutionRawHistoryV2(GetWorkflowExecutionRawHistoryV2Request) returns(GetWorkflowExecutionRawHistoryV2Response);
+
+  // GetReplicationMessages returns new replication tasks since the read level provided in the token.
+  rpc GetReplicationMessages(GetReplicationMessagesRequest) returns(GetReplicationMessagesResponse);
+
+  // GetDLQReplicationMessages return replication messages based on DLQ info.
+  rpc GetDLQReplicationMessages(GetDLQReplicationMessagesRequest) returns (GetDLQReplicationMessagesResponse);
+
+  // GetDomainReplicationMessages returns new domain replication tasks since last retrieved task id.
+  rpc GetDomainReplicationMessages (GetDomainReplicationMessagesRequest) returns (GetDomainReplicationMessagesResponse);
+
+  // ReapplyEvents applies stale events to the current workflow and current run.
+  rpc ReapplyEvents(ReapplyEventsRequest) returns (ReapplyEventsResponse);
+
+  // AddSearchAttribute whitelist search attribute in request.
+  rpc AddSearchAttribute (AddSearchAttributeRequest) returns (AddSearchAttributeResponse);
+
+  // DescribeCluster returns information about Cadence cluster.
+  rpc DescribeCluster(DescribeClusterRequest) returns (DescribeClusterResponse);
+
+  // ReadDLQMessages returns messages from DLQ.
+  rpc ReadDLQMessages(ReadDLQMessagesRequest) returns (ReadDLQMessagesResponse);
+
+  // PurgeDLQMessages purges messages from DLQ.
+  rpc PurgeDLQMessages(PurgeDLQMessagesRequest) returns (PurgeDLQMessagesResponse);
+
+  // MergeDLQMessages merges messages from DLQ.
+  rpc MergeDLQMessages(MergeDLQMessagesRequest) returns (MergeDLQMessagesResponse);
+
+  // RefreshWorkflowTasks refreshes all tasks of a workflow.
+  rpc RefreshWorkflowTasks(RefreshWorkflowTasksRequest) returns (RefreshWorkflowTasksResponse);
+
+  // ResendReplicationTasks requests replication tasks from remote cluster and apply tasks to current cluster.
+  rpc ResendReplicationTasks(ResendReplicationTasksRequest) returns (ResendReplicationTasksResponse);
+}
+
+message DescribeWorkflowExecutionRequest {
+  string domain = 1;
+  api.v1.WorkflowExecution workflow_execution = 2;
+}
+
+message DescribeWorkflowExecutionResponse {
+  string shard_id = 1;
+  string history_addr = 2;
+  string mutable_state_in_cache = 3;
+  string mutable_state_in_database = 4;
+}
+
+message DescribeHistoryHostRequest {
+  string host_address = 1;
+  int32 shard_id = 2;
+  api.v1.WorkflowExecution workflow_execution = 3;
+}
+
+message DescribeHistoryHostResponse {
+  int32 number_of_shards = 1;
+  repeated int32 shard_ids = 2;
+  shared.v1.DomainCacheInfo domain_cache = 3;
+  string shard_controller_status = 4;
+  string address = 5;
+}
+
+message CloseShardRequest {
+  int32 shard_id = 1;
+}
+
+message CloseShardResponse {
+}
+
+message RemoveTaskRequest {
+  int32 shard_id = 1;
+  shared.v1.TaskType task_type = 2;
+  int64 task_id = 3;
+  google.protobuf.Timestamp visibility_time = 4 [(gogoproto.stdtime) = true];
```

Comment:
What is this = true?

Reply:
`[(gogoproto.stdtime) = true]` -> an annotation.
in this case, it's to tell the gogoproto code generator to use the stdlib time type rather than something that directly fits the google.protobuf.Timestamp structure (which is basically a `struct{seconds int64, nanos int32}`)